Tests/Test_BSP_GPIO.c

```diff
@@ -0,0 +1,42 @@
+#include "common.h"
+#include "config.h"
+#include <bsp.h>
+
+int main(void) {
+
+    BSP_GPIO_Init(PORTA);
+    BSP_GPIO_Init(PORTB);
+
+    char port;
+    char rw;
+    uint16_t data = 0;
+
+    while (1) {
+        printf("Choose Port (A-D): ");
+        port = getchar();
+        getchar();
+
+        if (port - 'A' >= NUM_PORTS) {
```

Comment:
Shouldn't this be a while loop? Why would you continue if they enter an invalid port?